packages/@react-spectrum/s2/src/ActionBar.tsx

```diff
@@ -0,0 +1,227 @@
+import {useLocalizedStringFormatter} from '@react-aria/i18n';
+import {useResizeObserver} from '@react-aria/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+const slideIn = keyframes(`
+  from { transform: translateY(100%); opacity: 0 }
+  to { transform: translateY(0px); opacity: 1 }
+`);
+
+const slideOut = keyframes(`
+  from { transform: translateY(0px); opacity: 1 }
+  to { transform: translateY(100%); opacity: 0 }
+`);
+
+const actionBarStyles = style({
+  borderRadius: 'lg',
+  backgroundColor: {
+    default: 'elevated',
+    isEmphasized: 'neutral'
+  },
+  boxShadow: 'elevated',
+  boxSizing: 'border-box',
+  outlineStyle: 'solid',
+  outlineWidth: 1,
+  outlineOffset: -1,
+  outlineColor: {
+    default: 'gray-200',
+    isEmphasized: 'transparent',
+    forcedColors: 'ButtonBorder'
+  },
+  paddingX: 8,
+  paddingY: 12,
+  display: 'flex',
+  gap: 16,
+  alignItems: 'center',
+  position: {
+    isInContainer: 'absolute'
+  },
+  bottom: 8,
+  insetStart: 8,
+  '--insetEnd': {
+    type: 'insetEnd',
+    value: 8
+  },
+  width: {
+    default: 'full',
+    isInContainer: 'auto'
+  },
+  marginX: 'auto',
+  maxWidth: 960,
+  zIndex: 3,
+  animation: {
+    isInContainer: slideIn,
+    isExiting: slideOut
+  },
+  animationDuration: 200
+});
+
+export interface ActionBarProps extends SlotProps {
+  isEmphasized?: boolean,
+  selectedItemCount?: number | 'all',
+  /** Handler that is called when the ActionBar clear button is pressed. */
+  onClearSelection?: () => void,
+  children: ReactNode,
+  scrollRef?: RefObject<HTMLElement | null>
+}
+
+export const ActionBarContext = createContext<ContextValue<Partial<ActionBarProps>, DOMRefValue<HTMLDivElement>>>(null);
+
+export const ActionBar = forwardRef(function ActionBar(props: ActionBarProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, ActionBarContext);
+  let domRef = useDOMRef(ref);
+
+  let isOpen = props.selectedItemCount !== 0;
+  let isExiting = useExitAnimation(domRef, isOpen && props.scrollRef != null);
+  if (!isOpen && !isExiting) {
+    return null;
+  }
+
+  return <ActionBarInner {...props} ref={domRef} isExiting={isExiting} />;
+});
+
+const ActionBarInner = forwardRef(function ActionBarInner(props: ActionBarProps & {isExiting: boolean}, ref: ForwardedRef<HTMLDivElement | null>) {
+  let {isEmphasized, selectedItemCount = 0, children, onClearSelection, isExiting} = props;
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-spectrum/s2');
+
+  // Store the last count greater than zero so that we can retain it while rendering the fade-out animation.
+  let [lastCount, setLastCount] = useState(selectedItemCount);
+  if ((selectedItemCount === 'all' || selectedItemCount > 0) && selectedItemCount !== lastCount) {
+    setLastCount(selectedItemCount);
+  }
+  
+  let staticColor = isEmphasized ? 'black' as const : undefined;
```

Comment:
Somehow we need to make this swap to white in dark theme and I haven't figured out how to do that yet. We don't always have a way of reading the current theme via JS anymore, it's all in CSS.